datamol/mol.py

```diff
@@ -154,38 +156,64 @@ def to_neutral(mol: Chem.rdchem.Mol) -> Optional[Chem.rdchem.Mol]:
 
 
 def sanitize_mol(
-    mol: Chem.rdchem.Mol, charge_neutral: bool = False, sanifix: bool = True
+    mol: Chem.rdchem.Mol,
+    charge_neutral: bool = False,
+    sanifix: bool = True,
+    verbose: bool = True,
 ) -> Optional[Chem.rdchem.Mol]:
-    """Sanitize molecule and fix common errors.
+    """An augmented version of RDKit `sanitize=True`. It uses a
+    mol-SMILES-mol conversion to catch potential aromaticity errors
+    and try to fix aromatic nitrogen (using the popular sanifix4 script).
+    Optionally, it can neutralize the charge of the molecule.
+
+    Note #1: Only the first conformer (if present) will be preserved and
+    a warning will be displayed if more than one conformer is detected.
 
-    Warning:
-        The procedure includes a SMILES conversion to avoid accasional aromaticity
-        errors. In consequence, all the properties and the conformers will be lost.
+    Note #2: The molecule's properties will be preserved but the atom's
+    properties will be lost.
 
     Args:
         mol: a molecule.
         charge_neutral: whether charge neutralization should be applied.
         sanifix: whether to run the sanifix from James Davidson
             (sanifix4.py) that try to adjust aromatic nitrogens.
+        verbose: Whether displaying a warning about multiple conformers.
 
     Returns:
         mol: a molecule.
     """
     if mol is None:
         return mol
 
+    # Extract properties.
+    original_mol = copy_mol(mol)
+    properties = original_mol.GetPropsAsDict()
+
     if charge_neutral:
         mol = to_neutral(mol)
 
     if sanifix:
         mol = _sanifix4.sanifix(mol)
 
-    if mol:
+    if mol is not None:
+
+        # Detect multiple conformers
+        if verbose and mol.GetNumConformers() > 1:
+            logger.warning(
+                f"The molecule contains multiple conformers. Only the first one will be preserved."
+            )
+
+        # Try catch to avoid occasional aromaticity errors
         try:
-            # Try catch to avoid occasional aromaticity errors
-            return to_mol(dm.to_smiles(mol), sanitize=True)  # type: ignore
+            # `cxsmiles` is used here to preserve the first conformer.
```

Comment:
```python
mol = dm.to_mol("CC(=[O+])OC1=NC=C2C=NNC2=C1C([O-])=O")  
#can also try with explicit hydrogens, does not change results: add_hs=True, explicit_only=False

mol = dm.conformers.generate(mol)  # Fig A
```
 ![image](https://user-images.githubusercontent.com/5290110/121571157-a7d4b080-c9f0-11eb-93cb-92f0d10c5d8b.png)

```
new_mol = sanitize_mol(mol, charge_neutral=True)  # Fig B

# charges are replaced by radical. This is minor and probably fine, but we might consider using hydrogen to fill the charge difference for `to_neutral` instead, although that creates new issues with copying conformers.
```
![image](https://user-images.githubusercontent.com/5290110/121571232-bc18ad80-c9f0-11eb-8e2b-527026dc250e.png)

Hydrogen atoms are missing. And if we attempt to try to get them back: 

```python
Chem.AddHs(new_mol, explicitOnly=True, addCoords=True)
```
![image](https://user-images.githubusercontent.com/5290110/121571299-d3f03180-c9f0-11eb-83b7-29e71532dec4.png)
Hydrogen positions cannot be completely recovered.

Reply:
> # charges are replaced by radical. This is minor and probably fine, but we might consider using hydrogen to fill the charge difference for `to_neutral` instead, although that creates new issues with copying conformers.

Dealing with mol with/without hydrogens is a pain by experience (I discovered that while working on berth). Just detecting whether a molecule contains noH/allH/explicitH is tricky! It means that you can't really add a `Chem.AddHs` into `to_neutral` without potentially modifying the user's input.

Unless you have a fix in mind I will likely ignore this for now.